dll/win32/browseui/basebarsite.cpp

```diff
@@ -287,12 +287,11 @@ HRESULT STDMETHODCALLTYPE CBaseBarSite::TranslateAcceleratorIO(LPMSG lpMsg)
 {
     if (!fCurrentActiveBar)
     {
-        if (lpMsg)
+        if (lpMsg && TranslateMessage(lpMsg))
```

Comment:
Why is this fix needed? (that one and the similar one below)

Reply:
`TranslateAcceleratorIO` has to return `S_OK` if and only if the message was translated. There is some chances that the message isn't translated.

Reply:
See also `CInternetToolbar::TranslateAcceleratorIO`.

```cxx
HRESULT STDMETHODCALLTYPE CInternetToolbar::TranslateAcceleratorIO(LPMSG lpMsg)
{
    HRESULT hr = S_FALSE;

    if (fMenuBar)
        hr = IUnknown_TranslateAcceleratorIO(fMenuBar, lpMsg);
    if (hr == S_OK)
        return hr;

    if (fControlsBar)
        hr = IUnknown_TranslateAcceleratorIO(fControlsBar, lpMsg);
    if (hr == S_OK)
        return hr;

    if (fNavigationBar)
        hr = IUnknown_TranslateAcceleratorIO(fNavigationBar, lpMsg);
    if (hr == S_OK)
        return hr;

    return S_FALSE;
}
```